recipes-app/iot2050-efivarfs-helper/files/iot2050-efivarfs-helper.tmpl

```diff
@@ -0,0 +1,180 @@
+            print("Done.")
+        else:
+            print("ERROR: {} is not loaded!".format(mod_required))
+            exit_on_failure(message_prepare_fail)
+    except subprocess.CalledProcessError as e:
+        dump_subprocess_error(e)
+        exit_on_failure(message_prepare_fail, code=e.returncode)
+
+    # tee-supplicant is used to serve the trusted applications running in
+    # optee-os for accessing RPMB partition.
+    try:
+        print("Checking for the service tee-supplicant... ", end='')
+        cmd = 'systemctl is-active tee-supplicant.service'
+        result = subprocess.run(
+            cmd, shell=True, capture_output=True, check=True, text=True)
+
+        print("Done.")
+    except subprocess.CalledProcessError as e:
+        print("ERROR: service is inactive!")
+        exit_on_failure(message_prepare_fail)
+
+    # efivarfs must be mounted with read & write option.
+    try:
+        print("Checking for efivarfs mounting... ", end='')
+        cmd = 'findmnt {}'.format(EFIVARFS_MOUNT_POINT)
+        result = subprocess.run(
+            cmd, shell=True, capture_output=True, check=True, text=True)
+        efivarfs_mount_error = 'not mounted.'
+        for line in result.stdout.split('\n'):
+            if line.startswith(EFIVARFS_MOUNT_POINT):
+                mount_options = line.strip().split(' ')[-1].split(',')
+                if 'rw' in mount_options and 'ro' not in mount_options:
+                    efivarfs_mount_error = ''
+                else:
+                    efivarfs_mount_error = 'not mounted as read & write.'
+
+        if efivarfs_mount_error != '':
+            print("ERROR:\tefivarfs is {}".format(efivarfs_mount_error))
+            exit_on_failure(message_prepare_fail)
+
+        print("Done.")
+    except subprocess.CalledProcessError as e:
+        dump_subprocess_error(e)
+        exit_on_failure(message_prepare_fail, code=e.returncode)
+
+    # efitools provides convenient commands to manipulate the UEFI authenticated
+    # variables.
+    try:
+        print("Checking for the efitools pre-installation... ", end='')
+        cmd = 'which efi-readvar'
+        result = subprocess.run(
+            cmd, shell=True, capture_output=True, check=True, text=True)
+
+        print("Done.")
+    except subprocess.CalledProcessError as e:
+        # if not installed, the `which` command returns non-zero
+        print("ERROR: not pre-installed!")
+        exit_on_failure(message_prepare_fail)
+
+    # By default the UEFI authenticated variable files in efivarfs have the
+    # immutable flag. This prevents from modifying it.
+    try:
+        change_efivarfs_immutable_flag(immutable=False)
+    except subprocess.CalledProcessError as e:
+        dump_subprocess_error(e)
+        exit_on_failure(message_prepare_fail, code=e.returncode)
+
+    print()
+    print('OK:\tNow you can use efitools to manipulate the UEFI authenticated variables')
```

Comment:
Hm, why not extend this tool to call efitools with the appropriate flags so that +i flag gets removed, efi variable is modified, and then +i flag gets set again?